### PR TITLE
[Sprint 3]Feature create note in cart

### PR DESCRIPTION
**Pull request by nguyenhuy95dn**
Feature create note in cart
Link redmine : https://dev.framgia.com/redmine/issues/114091
Task : Order
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartFragment.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public static NoteCartFragment newInstance(CartItem cartItem) {
+        NoteCartFragment fragment = new NoteCartFragment();
+        Bundle args = new Bundle();
+        args.putParcelable(KEY_CART_ITEM, cartItem);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartFragment.java]: remove this method if unused

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartPresenter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Override
+    public void onStart() {
+        mProductRepository.openTransaction();
+    }
+
+    @Override
+    public void onStop() {
+        mCompositeSubscription.clear();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartPresenter.java]: Please add ``mProductRepository.closeTransaction()``

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartPresenter.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Override
+    public void editNoteCart(CartItem cartItem) {
+        if (cartItem == null) {
+            return;
+        }
+        Subscription subscriptions =
+                mProductRepository.editNoteCart(cartItem.getProductId(), cartItem.getNotes())
+                        .subscribe(new Subscriber<Void>() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartPresenter.java]: replace by 
```java
subscribe()
```

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartViewModel.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * Exposes the data to be used in the Notecart screen.
+ */
+
+public class NoteCartViewModel extends BaseObservable implements NoteCartContract.ViewModel {
+    private static final String TAG = "NoteCartViewModel";
+    private NoteCartContract.Presenter mPresenter;
+
+    @Bindable
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartViewModel.java]: move this line to above this line [62](https://github.com/framgia/android_forder_01/pull/224/files#diff-c5bad5b9b8495c799dcbf30cb0a51f56R62)

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartContract.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * This specifies the contract between the view and the presenter.
+ */
+interface NoteCartContract {
+    /**
+     * View.
+     */
+    interface ViewModel extends BaseViewModel<Presenter> {
+        void onEditNoteSuccess();
+
+        void onEditNoteError(Throwable e);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartContract.java]: Please remove these methods

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartViewModel.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    public void onStop() {
+        mPresenter.onStop();
+    }
+
+    @Override
+    public void setPresenter(NoteCartContract.Presenter presenter) {
+        mPresenter = presenter;
+    }
+
+    public TextWatcher getNoteWatcher() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartViewModel.java]: Please remove this method

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/SimpleTextWatcher.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package com.framgia.forder.screen.cart.notecart;
+
+import android.support.annotation.NonNull;
+import android.text.Editable;
+import android.text.TextWatcher;
+
+public class SimpleTextWatcher implements TextWatcher {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/SimpleTextWatcher.java]: remove this class

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartPresenter.java (hunk 2 above)
Comment (nguyenhuy95dn) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartPresenter.java]: added

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartViewModel.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    public void onStop() {
+        mPresenter.onStop();
+    }
+
+    @Override
+    public void setPresenter(NoteCartContract.Presenter presenter) {
+        mPresenter = presenter;
+    }
+
+    public View.OnFocusChangeListener getNoteFocusChangeListener() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartViewModel.java]: remove this line

File: app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartViewModel.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            public void onFocusChange(View view, boolean hasFocus) {
+                if (!hasFocus) {
+                    notifyPropertyChanged(BR.cartItem);
+                }
+            }
+        };
+    }
+
+    @Bindable
+    public CartItem getCartItem() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/notecart/NoteCartViewModel.java]: remove this method

**Pull request merged**